Backpack/src/main/java/net/skyscanner/backpack/calendar2/BpkCalendar.kt

```diff
@@ -73,83 +81,38 @@ class BpkCalendar private constructor(
     }
 
   init {
-    val headerHeight = ResourcesUtil.dpToPx(50, context)
-    initHeaderView(headerHeight)
-    initRecyclerView(headerHeight)
-    initYearView(headerHeight)
-  }
-
-  fun addOnScrollListener(listener: CalendarOnScrollListener) {
-    scrollListeners += listener
-  }
-
-  fun removeOnScrollListener(listener: CalendarOnScrollListener) {
-    scrollListeners -= listener
-  }
-
-  private fun initHeaderView(headerHeight: Int) {
-    val headerView = CalendarHeaderView(context)
-
-    addView(headerView) {
-      width = LayoutParams.MATCH_PARENT
-      height = headerHeight
-    }
+    recyclerView.layoutManager = calendarLayoutManager
+    recyclerView.adapter = calendarAdapter
+    recyclerView.itemAnimator = null
+    recyclerView.addOnScrollListener(object : RecyclerView.OnScrollListener() {
+      override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
 
-    state.onEach {
-      headerView(it.params)
-    }.launchIn(scope)
-  }
+        val firstItemPosition = calendarLayoutManager.findFirstVisibleItemPosition()
+        val item = state.value.cells[firstItemPosition]
 
-  private fun initRecyclerView(headerHeight: Int) {
-    val calendarSpanSizeLookup = CalendarSpanSizeLookup()
-    val calendarLayoutManager = CalendarLayoutManager(context, calendarSpanSizeLookup)
-
-    val recyclerView = RecyclerView(context).apply {
-      layoutManager = calendarLayoutManager
-      adapter = calendarAdapter
-      itemAnimator = null
-
-      addOnScrollListener(object : RecyclerView.OnScrollListener() {
-        override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
-
-          val firstItemPosition = calendarLayoutManager.findFirstVisibleItemPosition()
-          val item = state.value.cells[firstItemPosition]
-
-          scrollListeners.forEach {
-            it.invoke(item.yearMonth)
-          }
+        scrollListeners.forEach {
```

Comment:
i actually had a look and the scroll listener isn't used at all right now, so maybe we don't need to add it until somebody needs it and we know what they actually need

Reply:
I'm not sure if we should build features that aren't actually needed and part of the design system. i'd rather keep it simple and then add what is needed later on. looking at the component in figma it looks like we don't have a footer support right now (but there is a legend at the top that the new version doesn't have)